ecolex/legislation.py

```diff
@@ -134,18 +114,27 @@ def harvest_file(upfile):
             'type': LEGISLATION,
             'legLanguage_es': set(),
             'legLanguage_fr': set(),
+            'legKeyword_en': [],
+            'legKeyword_fr': [],
+            'legKeyword_es': [],
+            'legSubject_en': [],
+            'legSubject_fr': [],
+            'legSubject_es': [],
         }
 
         for k, v in FIELD_MAP.items():
-            field_values = get_content(document.findAll(META, {'name': k}))
+            field_values = get_content(document.findAll(k))
 
             if field_values and v not in MULTIVALUED_FIELDS:
                 field_values = field_values[0]
 
-            # if v in DATE_FIELDS and field_values:
-            #    field_values = get_date_format(field_values)
-
             if field_values:
+                if v == 'legKeyword_code':
+                    if ('legMainKeyword_code' in legislation and
+                            'legMainKeyword_code' in field_values):
+                        field_values.insert(0, field_values.pop(
+                                field_values.index(legislation.get('legMainKeyword_code'))))
+                        del legislation['legMainKeyword_code']
```

Comment:
This is a bit difficult to read. Try rewriting it. Here's a suggestion, although I'm sure it can be done even better:
```
if (v == 'legKeyword_code' and 
        'legMainKeyword_code' in legislation and
        'legMainKeyword_code' in field_values):
    kw_code = legislation.pop('legMainKeyword_code')
    field_values.insert(0, field_values.pop(field_values.index(kw_code)))
```